modules/machineset-yaml-rhv.adoc

```diff
@@ -0,0 +1,108 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating-infrastructure-machinesets.adoc
+// * machine_management/creating_machinesets/creating-machineset-rhv.adoc
+
+[id="machineset-yaml-rhv_{context}"]
+=  Sample YAML for a machine set custom resource on {rh-virtualization}
+
+This sample YAML defines a machine set that runs on {rh-virtualization} and creates nodes that are labeled with `node-role.kubernetes.io/<node_role>: ""`.
+
+In this sample, `<infrastructure_id>` is the infrastructure ID label that is based on the cluster ID that you set when you provisioned the cluster, and `<role>` is the node label to add.
+
+[source,yaml,subs="+quotes"]
+----
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: <infrastructure_id> <1>
+    machine.openshift.io/cluster-api-machine-role: <role> <2>
+    machine.openshift.io/cluster-api-machine-type: <role> <2>
+  name: <infrastructure_id>-<role> <3>
```

Comment:
@stoobie I believe the max size for the name is 35 chars to avoid: https://bugzilla.redhat.com/show_bug.cgi?id=1915122

Example of max chars in a hostname:  openshift-stage-wz4zh-infra-0-j6jsw (35 characters)

@vagnerfarias do you mind to share your thoughts here?

Reply:
As we could observe in one environment, if the machineset name is greater than 35 characters, it'll break mDNS for IPI deployments. We should include some warning in the documentation so that operators are aware of this. Aforementioned bz has more details on the issue.

Reply:
> As we could observe in one environment, if the machineset name is greater than 35 characters, it'll break mDNS for IPI deployments. We should include some warning in the documentation so that operators are aware of this. Aforementioned bz has more details on the issue.

@vagnerfarias Does the 35 character limitation *only* apply to *Installing a cluster quickly on RHV* (I believe this is the section on IPI). For UPI, does the limitation *not* apply?

In any case, I'm not sure it would be helpful to make this distinction in the documentation--will the person that installed OCP be the same person to create the machine set? Will whoever is creating the machine set know with certainty which installation method was used?

Reply:
It only applies to IPI for on-prem installations, including RHV, OSP, vSphere and bare metal. Corresponding RHV documentation is  _Installing a cluster quickly on RHV_ and _Installing a cluster on RHV with customizations_. 

I'm not sure there's an easy way to identify which installation method was used, so your concern is valid IMO.

Reply:
It is not clear that an admin would know which method was used for the cluster deployment.  Ideally, it should not matter whether IPI or UPI was used. However, in this particular case, it will break the cluster mDNS if you exceed the length of the cluster name.
I think we can call out the difference between of the cluster is using it's own external DNS (UPI) vs. the internal mDNS.  @dougsland Is there an easy way the admin can tell if the mDNS service is being used, vs. an external one?

Reply:
Hey @peterclauterbach I would assume a good practice for both deploys using short number of chars. No need to detect the type of installation in IMO. Sounds a good plan?

@StevenTobin ^